library/Makefile

```diff
@@ -14,7 +14,7 @@ _LIBS := $(shell find . -mindepth 2 -name Makefile -exec dirname {} \;  | sort)
 
 FORCE:
 $(_LIBS): FORCE
-	$(MAKE) -C $@ $(TARGET)
+	$(MAKE) -C $@
```

Comment:
This is incorrect. `$(TARGET)` is a target-specific variables and is set on line 19. it also default to an empty string to build the default target.

Reply:
You are correct, sorry.
The issue was the `lib: TARGET:=""` which caused make (version 4.3) to throw the error:
```
make: *** empty string invalid as file name.  Stop.
```

Reply:
> The issue was the `lib: TARGET:=""` which caused make (version 4.3) to throw the error

sorry about that. 

should I still make the `flock` change?

Reply:
Yes, we need to remove the component.xml file if the build fails.

Reply:
`.DELETE_ON_ERROR:` worked, thanks